src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> strategiesMap;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> strategiesMap) {
+        this.strategiesMap = strategiesMap;
+    }
+
+    @Override
+    public OperationHandler getHandler(Operation operation) {
+        String exceptionMessage =
+                "Cannot find OperationHandler for operation = [" + operation + "]";
+        return Optional.ofNullable(strategiesMap.get(operation))
+                .orElseThrow(() -> new NoSuchElementException(exceptionMessage));
```

Comment:
Consider using a more specific runtime exception rather than NoSuchElementException. NoSuchElementException is typically used for iteration beyond the end of a collection, and in this context, a more informative custom exception or IllegalArgumentException might be more appropriate, as it indicates that the given operation is not supported.